src/Utilities/FileUtilities.cs

```diff
@@ -310,5 +311,28 @@ public static string PathRelativeTo(string path, string relativeToDirectory)
             Debug.Assert(string.Compare(Path.GetFullPath(Path.Combine(relativeToDirectory, returnVal)), fullPath, StringComparison.OrdinalIgnoreCase) == 0);
             return returnVal;
         }
+
+        /// <summary>
+        /// Returns the file name and extension of the specified path string.
+        /// Path.GetFileName will not trim the file name from a Windows path 
+        /// when running on Unix because `\` is a legal file char on Unix.
+        /// See the Remarks section in the docs: https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getfilename
+        /// This method ignores that logic and provides the string after the last '\' when running on non-Windows platform and Path.GetFileName returns the same string.
+        /// </summary>
+        public static string GetPlatformIndependentFileName(string path)
```

Comment:
We should mark this as internal so that we minimize the API surface that we expose.  One could argue that this belongs in the core libraries.

Reply:
I believe that's handled by the `#if UTILITIES_PUBLIC` around the whole FileUtilities class. It's not publicly exposed from TraceEvent because TraceEvent.csproj doesn't define UTILITIES_PUBLIC.